### PR TITLE
rounding functions

### PR DESCRIPTION
**Pull request by izulin**
### Context

5 new functions and 1 new alias.
Addresses #582 
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1.
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [x] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: test/interpreter/function-ceiling.math.spec.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    expect(engine.getCellValue(adr('A8'))).toEqual(0)
+  })
+
+  it('should work with mode for negative numbers', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=CEILING.MATH(-11, -2)'],
+      ['=CEILING.MATH(-11, -2, 0)'],
+      ['=CEILING.MATH(-11, -2, 1)'],
+      ['=CEILING.MATH(-11, 0, 1)'],
+      ['=CEILING.MATH(-11, 0, 0)'],
```
Comment (wojciechczerniak) [inline test/interpreter/function-ceiling.math.spec.ts:57]: Excel docs are wrong again? "By default, significance is +1 for positive numbers and -1 for negative numbers."

https://support.microsoft.com/en-us/office/ceiling-math-function-80f95d2f-b499-4eee-9f16-f795a8e306c8

File: test/interpreter/function-ceiling.spec.ts [hunk 2]
```diff
@@ -1,95 +1,59 @@
...
-
-    expect(engine.getCellValue(adr('A1'))).toEqual(-10)
-  })
-})
+import {ErrorType, HyperFormula} from '../../src'
+import {ErrorMessage} from '../../src/error-message'
+import {adr, detailedError} from '../testUtils'
+
+describe('Function CEILING', () => {
+  /*Inconsistent with ODDF standard.*/
```
Comment (wojciechczerniak) [inline test/interpreter/function-ceiling.spec.ts:6]: Interesting, so the ODDF compilant is the CEILING.MATH in Excel?

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

File: docs/guide/built-in-functions.md [hunk 3]
```diff
@@ -183,7 +183,9 @@ lets you design your own [custom functions](custom-functions).
 | ATAN2 | Math and trigonometry | Returns the inverse trigonometric tangent of the specified x and y coordinates. | ATAN2(Numberx; Numbery) |
 | ATANH <br><Badge text="v0.2.0"/>| Math and trigonometry | Returns the inverse harmonic tangent of a number. | ATANH(Number) |
 | BASE | Math and trigonometry | Converts a positive integer to a specified base into a text from the numbering system. | BASE(Number; Radix; [Minimumlength]) |
-| CEILING | Math and trigonometry | Rounds a number up to the nearest multiple of Significance. | CEILING(Number; Significance; Mode) |
+| CEILING | Math and trigonometry | Rounds a number up to the nearest multiple of Significance. | CEILING(Number; Significance) |
```
[the remark this answers is not in the document]
Comment (wojciechczerniak) [reply docs/guide/built-in-functions.md:186]: @scarletfog How do we mark breaking changes according to our changelog format?

File: test/interpreter/function-ceiling.spec.ts (hunk 2 above)
Comment (izulin) [reply test/interpreter/function-ceiling.spec.ts:6]: yes

File: docs/guide/built-in-functions.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (scarletfog) [reply docs/guide/built-in-functions.md:186]: @wojciechczerniak not sure how we can mark it in this file - another label maybe?

But in the changelog file, we usually write such changes at the very top, in the "breaking changes" section.

[the remark this answers is not in the document]
Comment (wojciechczerniak) [reply docs/guide/built-in-functions.md:186]: I think that Keep Log "Changed" category is to mark changes that are not backward compatibile.

File: test/interpreter/function-ceiling.math.spec.ts (hunk 1 above)
Comment (izulin) [reply test/interpreter/function-ceiling.math.spec.ts:57]: wording is weird, but it's actually consistent

Comment (wojciechczerniak) [reply test/interpreter/function-ceiling.math.spec.ts:57]: Ok, so they do not mention the actual mode value, just how it will work by default 🤔

Comment (izulin) [reply test/interpreter/function-ceiling.math.spec.ts:57]: if you look at our implementation, we actually also set significance to be +1 or -1 depending on input value

File: docs/guide/built-in-functions.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (izulin) [reply docs/guide/built-in-functions.md:186]: added into the log

File: test/interpreter/function-floor.spec.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import {ErrorType, HyperFormula} from '../../src'
+import {ErrorMessage} from '../../src/error-message'
+import {adr, detailedError} from '../testUtils'
+
+describe('Function FLOOR', () => {
+  /*Inconsistent with ODDF standard.*/
```
Comment (bardek8) [inline test/interpreter/function-floor.spec.ts:6]:
```suggestion
  /*Inconsistent with ODFF standard.*/
```

File: test/interpreter/function-floor.spec.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(4.2)
+    expect(engine.getCellValue(adr('A2'))).toEqual(4.2)
+    expect(engine.getCellValue(adr('A3'))).toEqual(4)
+    expect(engine.getCellValue(adr('A4'))).toEqual(-1.8)
+    expect(engine.getCellValue(adr('A5'))).toEqual(0)
+    expect(engine.getCellValue(adr('A6'))).toEqual(0)
+  })
+
+  /*Inconsistent with ODDF standard.*/
```
Comment (bardek8) [inline test/interpreter/function-floor.spec.ts:45]:
```suggestion
  /*Inconsistent with ODFF standard.*/
```

File: test/interpreter/function-ceiling.spec.ts (hunk 2 above)
Comment (bardek8) [inline test/interpreter/function-ceiling.spec.ts:6]:
```suggestion
  /*Inconsistent with ODFF standard.*/
```

File: test/interpreter/function-ceiling.spec.ts [hunk 6]
```diff
@@ -1,95 +1,59 @@
...
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(4.5)
+    expect(engine.getCellValue(adr('A2'))).toEqual(4.8)
+    expect(engine.getCellValue(adr('A3'))).toEqual(6)
+    expect(engine.getCellValue(adr('A4'))).toEqual(-3.6)
+    expect(engine.getCellValue(adr('A5'))).toEqual(0)
+    expect(engine.getCellValue(adr('A6'))).toEqual(0)
+  })
+
+  /*Inconsistent with ODDF standard.*/
```
Comment (bardek8) [inline test/interpreter/function-ceiling.spec.ts:45]:
```suggestion
  /*Inconsistent with ODFF standard.*/
```

**Review by bardek8 — APPROVED** [review]
Looks good, thanks! 
I found only one place of inconsistency between XL/GS and ODFF, probably I would stick to XL/GS.

@wojciechczerniak , I'm passing the token back to you :)

File: test/interpreter/function-floor.spec.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      ['=FLOOR(-3.14, -1.8)'],
+      ['=FLOOR(-3.14, 0)'],
+      ['=FLOOR(3.14, 0)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(4.2)
+    expect(engine.getCellValue(adr('A2'))).toEqual(4.2)
+    expect(engine.getCellValue(adr('A3'))).toEqual(4)
+    expect(engine.getCellValue(adr('A4'))).toEqual(-1.8)
+    expect(engine.getCellValue(adr('A5'))).toEqual(0)
```
[the remark this answers is not in the document]
Comment (wojciechczerniak) [reply test/interpreter/function-floor.spec.ts:41]: Compatibility with ODFF is already broken for FLOOR and CEIL functions. If I understood correctly ODFF compliant are FLOOR.MATH, and CEIL.MATH. So we can go full GS/XL here. WDYT @izulin?

File: CHANGELOG.md [hunk 8]
```diff
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added 18 mathematical functions ROMAN, ARABIC, FACT, FACTDOUBLE, COMBIN, COMBINA, GCD, LCM, MROUND, MULTINOMIAL, QUOTIENT, RANDBETWEEN, SERIESSUM, SIGN, SQRTPI, SUMX2MY2, SUMX2PY2, SUMXMY2.
 - Added 63 statistical functions EXPON.DIST, EXPONDIST, FISHER, FISHERINV, GAMMA, GAMMA.DIST, GAMMADIST, GAMMALN, GAMMALN.PRECISE, GAMMA.INV, GAMMAINV, GAUSS, BETA.DIST, BETADIST, BETA.INV, BETAINV, BINOM.DIST, BINOMDIST, BINOM.INV, BESSELI, BESSELJ, BESSELK, BESSELY, CHISQ.DIST, CHISQ.DIST.RT, CHISQ.INV, CHISQ.INV.RT, CHIDIST, CHIINV, F.DIST, F.DIST.RT, F.INV, F.INV.RT, FDIST, FINV, WEIBULL, WEIBULL.DIST, HYPGEOMDIST, HYPGEOM.DIST, T.DIST, T.DIST.2T, T.DIST.RT, T.INV, T.INV.2T, TDIST, TINV, LOGNORM.DIST, LOGNORMDIST, LOGNORM.INV, LOGINV, NORM.DIST, NORMDIST, NORM.S.DIST, NORMSDIST, NORM.INV, NORMINV, NORM.S.INV, NORMSINV, PHI, NEGBINOM.DIST, NEGBINOMDIST, POISSON, POISSON.DIST.
 - Added function aliases mechanism. (#569)
+- Added 6 math function CEILING.MATH, FLOOR.MATH, FLOOR, CEILING.PRECISE, FLOOR.PRECISE, ISO.CEILING.
+
+### Changed
+- Changed CEILING function implementation to be consistent with existing implementations.
```
Comment (wojciechczerniak) [inline CHANGELOG.md:15]: Please add PR/Issue number

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]
All ok. We have to decide what to do with CEILING and FOOLR compatibility

File: test/interpreter/function-floor.spec.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (izulin) [reply test/interpreter/function-floor.spec.ts:41]: I'm all in for GS/XL compatibility.

@bardek8 but FLOOR(0,0) is not an error :)

**Review by wojciechczerniak — APPROVED** [review]
LGTM 👍

**Review by bardek8 — APPROVED** [review]
:+1:

**Pull request merged**